systemd/systemd.go

```diff
@@ -1091,9 +1143,28 @@ LazyUnmount=yes
 WantedBy=multi-user.target
 `
 
-func writeMountUnitFile(snapName, revision, what, where, fstype string, options []string) (mountUnitName string, err error) {
-	content := fmt.Sprintf(mountUnitTemplate, snapName, revision, what, where, fstype, strings.Join(options, ","))
-	mu := MountUnitPath(where)
+const (
+	snappyOriginModule = "X-SnapdOrigin"
+)
+
+func writeMountUnitFile(u *MountUnitOptions) (mountUnitName string, err error) {
+	if u == nil {
+		panic("writeMountUnitFile() expects valid mount options")
+	}
+
+	snapDesc := u.SnapName
```

Comment:
(nitpick^2) I wonder if text.Template would make things easier to read here - probably followup material and maybe/likely not really helpful but another one of my idle thoughts :)

Reply:
I'll take this suggestion in a follow-up PR, since the change is not trivial and I don't want to delay this too much :-)